perllib/FixMyStreet/App/Controller/Reports.pm

```diff
@@ -378,6 +378,18 @@ sub load_and_group_problems : Private {
         non_public => 0,
         state      => [ keys %$states ]
     };
+    my $filter = {
+        order_by => $c->stash->{sort_order},
+        rows => $c->cobrand->reports_per_page,
+    };
+
+    if (defined $c->stash->{filter_status}{shortlisted}) {
+        $where->{'me.id'} = { '=', \"user_planned_reports.report_id"};
+        $where->{'user_planned_reports.removed'} = { 'IS', \"NULL" };
+        $filter->{join} = 'user_planned_reports';
+    } elsif (defined $c->stash->{filter_status}{unshortlisted}) {
+        $where->{'me.id'} = { 'NOT IN', \"(SELECT problem.id FROM problem, user_planned_reports WHERE problem.id = user_planned_reports.report_id AND user_planned_reports.removed IS NULL)" };
```

Comment:
I like the approach of a subquery here. There's a `as_query` helper function you might want to use, as well as the `-not_in` operator, so you're leaning on the DB library instead of dropping to raw SQL. Something like
```
        my $shortlisted_ids = $c->cobrand->problems->search({
            'me.id' => { '=', \"user_planned_reports.report_id"},
            'user_planned_reports.removed' => undef,
        }, {
           join => 'user_planned_reports',
           columns => ['me.id'],
        })->as_query;
        $where->{'me.id'} = { -not_in => $shortlisted_ids };
```

should do the trick. There's some duplication of the query with the other clause in the parent `if`, so if you really wanted you could factor the common stuff out into a `shortlisted` sub in `ResultSet::Problem`, but 🤷‍♂️ .